numba/cuda/cudadrv/devicearray.py

```diff
@@ -663,3 +657,33 @@ def auto_device(obj, stream=0, copy=True):
             devobj.copy_to_device(obj, stream=stream)
         return devobj, True
 
+
+def squeezed_shape_and_strides(ary):
```

Comment:
I'm going to preempt the code review here and suggest myself that this be replaced with a `squeeze()` method on `DeviceNDArrayBase`.